quickwit-core/src/index.rs

```diff
@@ -99,11 +98,42 @@ pub async fn delete_index(
         .mark_splits_as_deleted(index_id, split_ids)
         .await?;
 
-    let files = garbage_collect(metastore.as_ref(), index_id, storage_resolver).await?;
+    let file_entries = garbage_remove(metastore.as_ref(), index_id, storage_resolver).await?;
     //TODO: discuss & fix possible data race
     metastore.delete_index(index_id).await?;
 
-    Ok(files)
+    Ok(file_entries)
+}
+
+/// Prepare all danglings files for removal from the index specified with `index_id`.
+/// It also performs the removal.
+///
+/// * `metastore_uri` - The metastore Uri for accessing the metastore.
+/// * `index_id` - The target index Id.
+///
+pub async fn garbage_collect_index(
+    metastore_uri: &str,
+    index_id: &str,
+) -> anyhow::Result<Vec<FileEntry>> {
+    let metastore = MetastoreUriResolver::default()
+        .resolve(&metastore_uri)
+        .await?;
+    let storage_resolver = StorageUriResolver::default();
+
+    // schedule all staged splits for delete
+    let staged_splits = metastore
+        .list_splits(index_id, SplitState::Staged, None)
+        .await?;
+    let split_ids = staged_splits
+        .iter()
+        .map(|split_meta| split_meta.split_id.as_str())
+        .collect::<Vec<_>>();
+    metastore
+        .mark_splits_as_deleted(index_id, split_ids)
+        .await?;
+
+    let file_entries = garbage_remove(metastore.as_ref(), index_id, storage_resolver).await?;
+    Ok(file_entries)
 }
 
 /// Removes all danglings files from an index specified at `index_uri`.
```

Comment:
Garbage_remove and garbage_collect do very different things, yet the name nor the comments are helping understand the difference.